rust/connlib/tunnel/src/dns.rs

```diff
@@ -217,48 +216,44 @@ impl StubResolver {
         ips
     }
 
-    // This function will panic if it's called with an invalid PTR question
-    fn get_records<N: ToName>(&mut self, question: &Question<N>) -> Vec<RecordData<DomainName>> {
+    fn get_records<N: ToName>(
+        &mut self,
+        question: &Question<N>,
+    ) -> Option<Vec<RecordData<DomainName>>> {
         match question.qtype() {
-            Rtype::A => self
-                .get_or_assign_ips(question.qname().to_name())
-                .iter()
-                .copied()
-                .filter_map(get_v4)
-                .map(domain::rdata::A::new)
-                .map(RecordData::A)
-                .collect_vec(),
-
-            Rtype::AAAA => self
-                .get_or_assign_ips(question.qname().to_name())
-                .iter()
-                .copied()
-                .filter_map(get_v6)
-                .map(domain::rdata::Aaaa::new)
-                .map(RecordData::Aaaa)
-                .collect_vec(),
-            Rtype::PTR => {
-                let Some(ip) = reverse_dns_addr(&question.qname().to_name::<Vec<_>>().to_string())
-                else {
-                    return Vec::new();
-                };
-                let Some(fqdn) = self.ips_to_fqdn.get(&ip) else {
-                    debug_assert!(false, "we expect this function to be called only with PTR records for resource ips");
-                    return Vec::new();
-                };
-
-                vec![RecordData::Ptr(domain::rdata::Ptr::new(fqdn.clone()))]
+            Rtype::A => {
+                get_description(&question.qname().to_name(), &self.dns_resources)?;
```

Comment:
This is pretty subtle. Can we perhaps check this outside and make this a (debug) assert instead?

I liked the previous `is_resource` check. I think that should be explicit in the code.

Reply:
Maybe we can split this function into three, one for each type so that it is more obvious in which cases we forward the query:

1. When it is for a non-resource
2. When it is for a type other than `A`, `AAAA` or `PTR`

If the function is split up, it shouldn't be too much boilerplate to have the `match` on level higher up. What do you think?

Reply:
I removed the `is_resource` to save some repetition on the forward query and also to save us the code for the special check for `PTR`.

I think rather than splitting this function into three, it'd be less repetition to bring back the `is_resource`

Reply:
Trying to hash something out that might make this clearer

Reply:
let me know what you think now